sdk/tables/Azure.Data.Tables/tests/TableClientQueryableLiveTests.cs

```diff
@@ -414,89 +414,94 @@ public async Task TableQueryableWithDictionaryTypeOnSupportedTypes()
+            results = await client.QueryAsync<TableEntity>(ent => ent.GetInt64("LongPrimitive") >= thirdEntity.LongPrimitive).ToEnumerableAsync().ConfigureAwait(false);
 
             Assert.That(results.Count, Is.EqualTo(2));
 
-            results = await client.QueryAsync(ent => (long?)ent["LongPrimitiveN"] >= thirdEntity.LongPrimitiveN).ToEnumerableAsync().ConfigureAwait(false);
+            results = await client.QueryAsync<TableEntity>(ent => ent.GetInt64("LongPrimitiveN") >= thirdEntity.LongPrimitiveN).ToEnumerableAsync().ConfigureAwait(false);
 
             Assert.That(results.Count, Is.EqualTo(2));
 
 
             // 4. Filter on Double
-            results = await client.QueryAsync(ent => (double)ent["Double"] >= thirdEntity.Double).ToEnumerableAsync().ConfigureAwait(false);
+            results = await client.QueryAsync<TableEntity>(ent => ent.GetDouble("Double") >= thirdEntity.Double).ToEnumerableAsync().ConfigureAwait(false);
 
             Assert.That(results.Count, Is.EqualTo(2));
 
-            results = await client.QueryAsync(ent => (double)ent["DoublePrimitive"] >= thirdEntity.DoublePrimitive).ToEnumerableAsync().ConfigureAwait(false);
+            results = await client.QueryAsync<TableEntity>(ent => ent.GetDouble("DoublePrimitive") >= thirdEntity.DoublePrimitive).ToEnumerableAsync().ConfigureAwait(false);
 
             Assert.That(results.Count, Is.EqualTo(2));
 
 
             // 5. Filter on Integer
-            results = await client.QueryAsync(ent => (Int32)ent["Int32"] >= thirdEntity.Int32).ToEnumerableAsync().ConfigureAwait(false);
+            results = await client.QueryAsync<TableEntity>(ent => ent.GetInt32("Int32") >= thirdEntity.Int32).ToEnumerableAsync().ConfigureAwait(false);
 
             Assert.That(results.Count, Is.EqualTo(2));
 
-            results = await client.QueryAsync(ent => (Int32?)ent["Int32N"] >= thirdEntity.Int32N).ToEnumerableAsync().ConfigureAwait(false);
+            results = await client.QueryAsync<TableEntity>(ent => ent.GetInt32("Int32N") >= thirdEntity.Int32N).ToEnumerableAsync().ConfigureAwait(false);
 
             Assert.That(results.Count, Is.EqualTo(2));
 
 
             // 6. Filter on Date
-            results = await client.QueryAsync(ent => (DateTimeOffset)ent["DateTimeOffset"] >= thirdEntity.DateTimeOffset).ToEnumerableAsync().ConfigureAwait(false);
+            results = await client.QueryAsync<TableEntity>(ent => (DateTimeOffset)ent["DateTimeOffset"] >= thirdEntity.DateTimeOffset).ToEnumerableAsync().ConfigureAwait(false);
 
             Assert.That(results.Count, Is.EqualTo(2));
 
-            results = await client.QueryAsync(ent => (DateTimeOffset)ent["DateTimeOffset"] < thirdEntity.DateTimeOffset).ToEnumerableAsync().ConfigureAwait(false);
+            results = await client.QueryAsync<TableEntity>(ent => (DateTimeOffset)ent["DateTimeOffset"] < thirdEntity.DateTimeOffset).ToEnumerableAsync().ConfigureAwait(false);
```

Comment:
@christothes also wondering if we should include a `GetDateTimeOffset` if it's a common use case in `TableEntity`.

Reply:
We probably should if you want to file a bug on it.

Reply:
Okay! I don't think there's other DateTimes to support, so tracking here: #14002